pull/github.go

```diff
@@ -289,6 +289,30 @@ func (ghc *GitHubContext) HasReveiwers() (bool, error) {
 	return len(subsetCurrentReviewers.Users) > 0 || len(subsetCurrentReviewers.Teams) > 0, nil
 }
 
+func (ghc *GitHubContext) Teams() (map[string]string, error) {
+	opt := &github.ListOptions{
+		PerPage: 100,
+	}
+
+	// get all pages of results
+	allTeams := make(map[string]string)
+	for {
+		teams, resp, err := ghc.client.Repositories.ListTeams(ghc.ctx, ghc.RepositoryOwner(), ghc.RepositoryName(), opt)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to list teams page %d", opt.Page)
+		}
+		for _, t := range teams {
+			allTeams[t.GetName()] = t.GetPermission()
+		}
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
+	}
+
+	return allTeams, nil
```

Comment:
Should this be cached like other list output in the context?

Reply:
We should get v3 caching by default, but I'm open to also caching it more explicitly here.

Reply:
Ok, we can try it with only the v3 caching for now.